pghoard/pghoard.py

```diff
@@ -171,6 +172,27 @@ def receivexlog_listener(self, site, connection_info, xlog_directory):
         thread.start()
         self.receivexlogs[site] = thread
 
+    def walreceiver(self, site, chosen_backup_node, last_flushed_lsn):
+        connection_string, slot = replication_connection_string_and_slot_using_pgpass(chosen_backup_node)
+        pg_version_server = self.check_pg_server_version(connection_string)
+        if pg_version_server:
+            self.config["backup_sites"][site]["pg_version"] = pg_version_server
+        if not self.check_pg_versions_ok(pg_version_server, "pg_receivexlog"):
+            return
+        # Imported here because WALReceiver requires an unreleased version of psycopg2
+        from pghoard.walreceiver import WALReceiver
```

Comment:
should still be imported at module level (except ImportError: WALReceiver = None)